lib/pg_rls/connection_config.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module PgRls
+  # The ConnectionConfig module provides methods to configure and manage
+  # the connection settings for Row Level Security (RLS) in PostgreSQL.
+  # It includes methods to look up and validate the connection configuration,
+  # as well as helper methods to build the configuration hash based on different modes.
+  module ConnectionConfig
+    class << self
```

Comment:
Can we make this a class that initialize with PgRls::ConnectionConfig.new(settings/connection/whatever).db_config or something like that to keep the same ActiveRecord pattern of invocation